common/src/main/java/org/opensearch/ml/common/MLModel.java

```diff
@@ -28,23 +35,74 @@ public class MLModel implements ToXContentObject {
     public static final String OLD_MODEL_CONTENT_FIELD = "content";
     public static final String MODEL_CONTENT_FIELD = "model_content";
 
+    public static final String DESCRIPTION_FIELD = "description";
+    public static final String MODEL_FORMAT_FIELD = "model_format";
+    public static final String MODEL_TASK_TYPE_FIELD = "model_task_type";
+    public static final String MODEL_STATE_FIELD = "model_state";
+    public static final String MODEL_CONTENT_SIZE_IN_BYTES_FIELD = "model_content_size_in_bytes";
+    public static final String MODEL_CONTENT_HASH_FIELD = "model_content_hash";
+    public static final String MODEL_CONFIG_FIELD = "model_config";
+    public static final String CREATED_TIME_FIELD = "created_time";
+    public static final String LATEST_UPLOADED_TIME_FIELD = "latest_uploaded_time";
+    public static final String LATEST_LOADED_TIME_FIELD = "latest_loaded_time";
+    public static final String LATEST_UNLOADED_TIME_FIELD = "latest_unloaded_time";
+
+    public static final String MODEL_ID_FIELD = "model_id";
+    public static final String CHUNK_NUMBER_FIELD = "chunk_number";
+    public static final String TOTAL_CHUNKS_FIELD = "total_chunks";
+
     private String name;
     private FunctionName algorithm;
```

Comment:
Based on the design, all custom models use "custom" in the algorithm field. We can use this algorithm field to check if model is build-in or custom. Is there any possibility in the future we want to distinguish different custom models by algorithms, so this field may need to include more custom model names? If we reserve this field to allow users to use different custom model algorithms, is it better to add a boolean field "isCustomModel" to help us identify if the model is build-in or custom?

Reply:
Good question. We can differentiate different custom model types with `MLModelTaskType`